src/material/datepicker/calendar.ts

```diff
@@ -69,11 +69,24 @@ export class MatCalendarHeader<D> {
     if (this.calendar.currentView == 'year') {
       return this._dateAdapter.getYearName(this.calendar.activeDate);
     }
+
```

Comment:
Could you break this multi-year part of the getter into its own function? The logic here has grown a bit large